code/modules/mob/dead/new_player/latejoin_menu.dm

```diff
@@ -0,0 +1,192 @@
+#define JOB_CHOICE_YES "Yes"
+#define JOB_CHOICE_REROLL "Reroll"
+#define JOB_CHOICE_CANCEL "Cancel"
+
+GLOBAL_DATUM_INIT(latejoin_menu, /datum/latejoin_menu, new)
+
+/// Makes a list of jobs and pushes them to a DM list selector. Just in case someone did a special kind of fucky-wucky with TGUI.
+/datum/latejoin_menu/proc/fallback_ui(mob/dead/new_player/user)
+	var/list/jobs = list()
+	for(var/datum/job/job in SSjob.joinable_occupations)
+		jobs += job.title
+
+	var/input_contents = input(user, "Pick a job to join as:", "Latejoin Job Selection") as null|anything in jobs
+
+	if(!input_contents)
+		return
+
+	user.AttemptLateSpawn(input_contents)
+
+/datum/latejoin_menu/ui_interact(mob/user, datum/tgui/ui)
+	ui = SStgui.try_update_ui(user, src, ui)
+	if(!ui)
+		ui = new(user, src, "JobSelection", "Latejoin Menu")
+		ui.open()
+
+/datum/latejoin_menu/ui_data(mob/user)
+	var/mob/dead/new_player/owner = user
+	var/list/departments = list()
+	var/list/data = list(
+		"disable_jobs_for_non_observers" = SSlag_switch.measures[DISABLE_NON_OBSJOBS],
+		"round_duration" = DisplayTimeText(world.time - SSticker.round_start_time),
+		"departments" = departments,
+	)
+	if(SSshuttle.emergency)
+		switch(SSshuttle.emergency.mode)
+			if(SHUTTLE_ESCAPE)
+				data["shuttle_status"] = "The station has been evacuated."
+			if(SHUTTLE_CALL)
+				if(!SSshuttle.canRecall())
+					data["shuttle_status"] = "The station is currently undergoing evacuation procedures."
+
+	for(var/datum/job/prioritized_job in SSjob.prioritized_jobs)
+		if(prioritized_job.current_positions >= prioritized_job.total_positions)
+			SSjob.prioritized_jobs -= prioritized_job
```

Comment:
This definitely shouldn't be here but if it already was then I'll just let it slide since I don't want to complicate this PR too much

Reply:
Yea, welcome to oldcode, and I guess I could add a getter for available jobs in SSJobs with that updating code inside, but that might be an extra step for something that folk already do themselves.